cleanlab_studio/internal/tlm/validation.py

```diff
@@ -17,6 +17,36 @@
 )
 
 
+def validate_tlm_prompt_kwargs_constrain_outputs(
+    prompt: Union[str, Sequence[str]],
+    **kwargs: Any,
+) -> Union[Optional[List[str]], Optional[List[Optional[List[str]]]]]:
+    # validate kwargs - only allow constrain_outputs
+    supported_kwargs = {"constrain_outputs"}
+    unsupported_kwargs = set(kwargs.keys()) - supported_kwargs
+    if unsupported_kwargs:
+        raise ValidationError(
+            f"Unsupported keyword arguments: {unsupported_kwargs}. "
+            f"Supported keyword arguments are: {supported_kwargs}"
+        )
+
+    # validate constrain_outputs and it needs to match with what prompt is whether prompt is a sequence or string
+    constrain_outputs = kwargs.get("constrain_outputs")
+    if constrain_outputs is not None:
+        if isinstance(prompt, str):
+            if not isinstance(constrain_outputs, list):
+                raise ValidationError("constrain_outputs must be a list of strings")
+        elif isinstance(prompt, Sequence):
+            if not isinstance(constrain_outputs, list) or not all(
+                isinstance(co, list) for co in constrain_outputs
+            ):
+                raise ValidationError("constrain_outputs must be a list of lists of strings")
```

Comment:
Did you need to verify that each item in the list is indeed a string? (seems like a lot to iterate through, but not sure if there will be weird failures later on if unverified)

Reply:
the backend will also verify so currently if the element is not a string it will get 400. But good point! It's better to verify here as well to raise error earlier. The for loop should be cheap so should be fine in term of compute cost.